### PR TITLE
Scexample

### PR DESCRIPTION
**Pull request by yramachers**
This example is now in an executable state for testing. The README gives some insight, could probably become more extensive.

### CODE REVIEW

**Comment by yramachers** [comment]
Ok, @TomGoffrey this is ready to go. All updates made and tested. There are now 10^7 events in 10 files on the SCRTP storage disk which you might be able to see (should have a read permission at /storage/epp/phsdaq/QTNM). These are for the baseline geometry with the physics lists option 3 and 4.

**Comment by yramachers** [comment]
For the record, I should point out that Multiple Scattering is the vastly dominant process with quite some explanations given in the Geant4 manual, here as an introduction to the various physics options:

https://geant4-userdoc.web.cern.ch/UsersGuides/ForApplicationDeveloper/html/TrackingAndPhysics/physicsProcess.html#electromagnetic-interactions

File: examples/ScatteringExample/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# Compiler/build settings
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+# CTest if we need it
+include(CTest)
+
+# Dependencies
+find_package(Geant4 REQUIRED)
```
Comment (TomGoffrey) [inline examples/ScatteringExample/CMakeLists.txt]: @yramachers Any objection to changing this to `find_package(Geant4 10.7 REQUIRED)`?

I'm fairly sure there's quite a few things (e.g. `#include "G4RunManagerFactory.hh"`) which require 10.7.

Comment (yramachers) [reply examples/ScatteringExample/CMakeLists.txt]: No, 10.7 really is a minimum requirement. Didn't know one can require a version of Geant4 in the Cmake. That's fine.

**Comment by yramachers** [comment]
Ok, cmake change made and it compiles fine on my SCRTP machine.
The next branch to make our own physics list is prepared. I mention that since I found a nice example using a new (from 10.7) single scattering model for electrons. Surprisingly, I had used it with the current version and saw no difference to the multiple scattering physics lists. Turns out, there is a bug in this new model code: two lines of code that shouldn't be there, i.e. they overwrite the logical switches for the model to decide between handling single scattering or multiple scattering (in favour of m.sc) hence no visible difference. I hope that the single scattering behaviour is better but haven't tried it yet. I'll hence make a patch version of that model for the new branch.

**Comment by TomGoffrey** [comment]
This also builds and runs fine on my (Mac) laptop. I can also confirm the change to cmake correctly stops if the Geant4 version is too low. Going to upgrade to 10.7 on my (Ubuntu) desktop, build & run there, then I'm happy to merge.

**Comment by TomGoffrey** [comment]
Builds and runs fine on Ubuntu too.

**Review by TomGoffrey — APPROVED** [review]

**Pull request merged**